third_party/move/move-prover/boogie-backend/src/prelude/prelude.bpl

```diff
@@ -24,6 +24,18 @@ options provided to the prover.
 {% include "custom-natives" %}
 {%- endif %}
 
+
+// Uninterpreted function for all types
+
+{% for instance in uninterpreted_instances %}
+
+{%- set S = "'" ~ instance.suffix ~ "'" -%}
+{%- set T = instance.name -%}
+
+function $Uninterpreted_value_of_{{S}}(): {{T}};
```

Comment:
The naming is a bit of off, because "uninterpreted value" does not make a lot of sense in contrast to "uninterpreted function". Perhaps call this "$Arbitrary_value_of"